web/src/message_list_data.js

```diff
@@ -219,7 +220,10 @@ export class MessageListData {
         }
 
         return messages.filter((message) => {
-            if (message.type !== "stream") {
+            if (
+                message.type !== "stream" ||
+                user_topics.is_topic_unmuted(message.stream_id, message.topic)
+            ) {
                 return true;
```

Comment:
This function claims to only filter for topic mutes; is it correct to change it?

Reply:
We no longer want to make any changes to this function with a new approach